yarn-project/p2p/src/mem_pools/tx_pool/README.md

```diff
@@ -0,0 +1,241 @@
+# Transaction Pool (Mempool)
+
+This module implements the transaction pool (mempool) for the Aztec P2P network. The mempool holds unconfirmed transactions awaiting inclusion in a block.
+
+## Overview
+
+The transaction pool serves as a staging area for transactions before they are included in blocks. It manages the lifecycle of transactions from initial submission through mining, handling duplicates, priority ordering, and eviction of invalid or low-priority transactions.
+
+## Interface: `TxPool`
+
+The [`TxPool`](tx_pool.ts) interface defines the contract that all transaction pool implementations must fulfill:
+
+### Transaction Lifecycle
+
+| Method | Description |
+|--------|-------------|
+| `addTxs(txs, opts?)` | Adds transactions to the pool. Duplicates are ignored. Returns count of newly added txs. |
+| `deleteTxs(txHashes, opts?)` | Removes transactions from the pool. Supports soft-delete for mined txs. |
+| `markAsMined(txHashes, blockHeader)` | Marks transactions as included in a block. |
+| `markMinedAsPending(txHashes, blockNumber)` | Reverts mined transactions to pending (used during reorgs). |
+| `getArchivedTxByHash(txHash)` | Retrieves archived (historical) transactions. |
+| `getTxStatus(txHash)` | Returns status: `'pending'`, `'mined'`, `'deleted'`, or `undefined`. |
+
+### Transaction Fetching
+
+| Method | Description |
+|--------|-------------|
+| `hasTx(txHash)` / `hasTxs(txHashes)` | Checks if transaction(s) exist in the pool. |
+| `getTxByHash(txHash)` | Retrieves a transaction by its hash. |
+| `getTxsByHash(txHashes)` | Batch retrieval of transactions by hash. |
+| `getAllTxs()` / `getAllTxHashes()` | Returns all transactions or their hashes. |
+| `getPendingTxHashes()` | Returns pending tx hashes **sorted by priority** (highest first). |
+| `getPendingTxCount()` | Returns count of pending transactions. |
+| `getMinedTxHashes()` | Returns mined tx hashes with their block numbers. |
+
+### Pool Management
+
+| Method | Description |
+|--------|-------------|
+| `updateConfig(config)` | Updates pool configuration (max size, archive limit). |
+| `markTxsAsNonEvictable(txHashes)` | Protects transactions from eviction. |
+| `clearNonEvictableTxs()` | Clears non-evictable flag from all transactions. |
+| `cleanupDeletedMinedTxs(blockNumber)` | Permanently removes soft-deleted txs from blocks ≤ blockNumber. |
+| `isEmpty()` | Checks if the pool has no transactions. |
+
+### Events
+
+The pool emits a `txs-added` event when new transactions are successfully added, allowing subscribers to react to pool changes.
+
+## Implementations
+
+### `InMemoryTxPool` (Testing Only)
```

Comment:
This reminds me that we should probably kill it, and just have a single implementation